mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -610,6 +610,28 @@ LogicalResult SYCLHostScheduleKernel::verify() {
   return verifyNdRange(*this, range, offset, ndRange);
 }
 
+void SYCLHostScheduleKernel::getEffects(
+    SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
+        &effects) {
+  auto *defaultResource = SideEffects::DefaultResource::get();
+  // The nd-range arguments will be read from
+  if (Value range = getRange())
+    effects.emplace_back(MemoryEffects::Read::get(), range, defaultResource);
+  if (Value offset = getOffset())
+    effects.emplace_back(MemoryEffects::Read::get(), offset, defaultResource);
+  // The rest of the arguments will be scalar or have rw access
+  for (Value arg : getArgs()) {
+    if (!isa<LLVM::LLVMPointerType>(arg.getType())) {
```

Comment:
This gives rw access to scalars and sets nothing for pointers; is the negation here intentional?

Reply:
Ups, nope, it's the other way aroun